core/network/besu/contracts/interop/manageAssetAny.sol

```diff
@@ -87,18 +135,20 @@ contract AssetExchangeContract is ERC1155Holder {
             expirationTime > block.timestamp,
             "Expiration time should be in the future"
         );
-
+        address receiver = bytesToAddress(fromHex(params.recipient));
         // The identity of the lock contract is a hash of all the relevant parameters that will uniquely identify the contract
-        lockContractId = sha256(
-            abi.encodePacked(
-                sender,
-                receiver,
-                assetContract,
-                amount,
-                hashLock,
-                expirationTime
-            )
-        );
+        if (hashMechanism == HashMechanism.SHA256) {
+            lockContractId = sha256(
```

Comment:
to generate lockContractID, you don't need to depend upon HashMechanism. HashMechanism is just for Hash Algo used in HTLC hash generation.
This lockContractID is independent of that and is just for book-keeping. So you can always have sha256 for this and remove the if condition check here.